ws/handler.go

```diff
@@ -114,3 +116,32 @@ func FetchUID(r *http.Request) (string, error) {
 
 	return requestID, nil
 }
+
+func CheckOrigin(origins string) func(r *http.Request) bool {
+	var hosts []string
+	if origins != "" {
+		hosts = strings.Split(strings.ToLower(origins), ",")
+	}
+
+	return func(r *http.Request) bool {
+		if len(hosts) == 0 {
```

Comment:
Maybe, we can move this check up and return a simple truthy function? (So, we don't check for the length every time)